packages/sqlite-sync/src/index.ts

```diff
@@ -0,0 +1,145 @@
+import {escapeSQLiteIdentifier} from '@databases/escape-identifier';
+import sql, {SQLQuery, isSqlQuery, FormatConfig} from '@databases/sql';
+import DatabaseConstructor, {Statement, Options as DatabaseOptions, Database} from 'better-sqlite3';
+
+export type {SQLQuery};
+export {sql, isSqlQuery};
+
+const sqliteFormat: FormatConfig = {
+  escapeIdentifier: (str) => escapeSQLiteIdentifier(str),
+  formatValue: (value) => ({placeholder: '?', value}),
+};
+
+export interface DatabaseTransaction {
+  query(query: SQLQuery): any[];
+  queryStream(query: SQLQuery): IterableIterator<any>;
+}
+export interface DatabaseConnection extends DatabaseTransaction {
+  tx<T>(fn: (db: DatabaseTransaction) => Promise<T>): Promise<T>;
```

Comment:
```suggestion
  tx<T>(fn: (db: DatabaseTransaction) => T): T;
```

Reply:
The reason why I kept the promise-based transaction API is because users will inevitably pass an `async` function here because they want to perform some other asynchronous operation and keep the database locked.

I'm happy to remove it... but I think this provides better ergonomics.